Automattic/wp-calypso
# Adds recordTracks option to the Promote Post widget

We want to unify how the Promote post widget handles tracks. To achieve that, we will give the widget a new wrapper function with the Calypso record track logic.

## Proposed Changes

* Adds a new Promote post widget parameter with a record track function
* Whitelist events generated by the widget in our Calypso tracks library

## Testing Instructions

**Note:** The widget will start to use the new recordTracks function starting from version `0.5.58`. You can check what version of the widget you are running by checking in the console for this message: `🪿🪿🪿 Welcome to BlazePress v0.5.58 🪿🪿🪿`.

* Open your browser network tab, and filter by `t.gif` calls
* Go to Tools->Advertising
* Click on the Promote button
* Check that the widget events are being correctly pushed. For example, you should be able to see new tracks events using the prefix `wpcom_dsp_widget_`. For example, the event `wpcom_dsp_widget_flow_start` is triggered at the start of the widget.

## Pre-merge Checklist
